src/libraries/Common/tests/System/FunctionPointerTests.cs

```diff
@@ -156,6 +156,11 @@ public static unsafe void FunctionPointerReturn()
             Assert.True(fcnPtr2.IsFunctionPointer);
 
             Assert.True(fcnPtr1.IsFunctionPointerEqual(fcnPtr2));
+
+            MethodInfo m3 = t.GetMethod(nameof(FunctionPointerHolder.GenericMethodReturnValue), Bindings);
+            Type fcnPtr3 = m3.ReturnType;
+            Assert.True(fcnPtr3.IsFunctionPointer);
+            Assert.True(fcnPtr3.ContainsGenericParameters);
```

Comment:
It would be good to add a test for FP generic arguments as well (in addition to return type here); can be done later.